src/main/java/oliver/Oliver.java

```diff
@@ -0,0 +1,205 @@
+     *
+     * @param input the user input provided
+     * @return string representation of Oliver's response
+     */
+    private String handleTodo(String input) {
+        try {
+            ToDo t = new ToDo(Parser.parseArgs(input));
+            tasks.add(t);
+            return ui.showAdd(t, tasks.getSize());
+        } catch (IndexOutOfBoundsException e) {
+            return ui.showMissingArgsError();
+        }
+    }
+
+    /**
+     * Adds a deadline task to the list.
+     *
+     * @param input the user input provided
+     * @return string representation of Oliver's response
+     */
+    private String handleDeadline(String input) { // Date is required, time is optional
+        try {
+            String[] parts = input.split("/by ");
+            String[] dateAndTime = parts[1].split(" ");
+
+            DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HHmm");
+            // Time set to default if no time is provided in the input
+            String timeStr = dateAndTime.length == 1 ? "0000" : dateAndTime[1];
+            LocalDate date = LocalDate.parse(dateAndTime[0]);
+            LocalTime time = LocalTime.parse(timeStr, timeFormatter);
+            LocalDateTime dateTime = LocalDateTime.of(date, time);
+
+            String action = parts[0].trim();
+            Deadline d = new Deadline(action.split(" ", 2)[1], dateTime);
+            tasks.add(d);
+            return ui.showAdd(d, tasks.getSize());
+        } catch (IndexOutOfBoundsException e) {
+            return ui.showMissingArgsError();
+        } catch (DateTimeParseException e) {
+            return "Invalid date or time. Please enter the date and time in the following format: " +
+                    "YYYY-MM-DD HHmm\nNote that date is required but time is optional.";
+        }
+    }
+
+    /**
+     * Adds an event task to the list.
+     *
+     * @param input the user input provided
+     * @return string representation of Oliver's response
+     */
+    private String handleEvent(String input) { // Both date and time are required
+        try {
+            String[] parts = input.split("/from |/to ");
+            String action = parts[0].trim();
+            String startDateAndTime = parts[1].trim();
+            String endDateAndTime = parts[2].trim();
+
+            DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+            LocalDateTime start = LocalDateTime.parse(startDateAndTime, dateFormatter);
+            LocalDateTime end = LocalDateTime.parse(endDateAndTime, dateFormatter);
+
+            Event e = new Event(action.split(" ", 2)[1], start, end);
+            tasks.add(e);
+            return ui.showAdd(e, tasks.getSize());
+        } catch (IndexOutOfBoundsException e) {
+            return ui.showMissingArgsError();
+        } catch (DateTimeParseException e) {
+            return "Invalid date or time. Please enter the date and time in the following format: YYYY-MM-DD HHmm";
```

Comment:
Beyond the soft limit of line length, you may want to consider adding a line break.